### PR TITLE
Add private ORDER_INFO subscription

### PR DESCRIPTION
**Pull request by KunduzB**
Make it possible to subscribe private ORDER_INFO data for Bitmex.

- Include ORDER_INFO channel to Bitmex
- Add parser for order message

### CODE REVIEW

File: cryptofeed/exchanges/bitmex.py [hunk 1]
```diff
@@ -484,6 +524,40 @@ async def _liquidation(self, msg: dict, timestamp: float):
...
+        'cum_exec_value': '0', 
+        'cum_exec_fee': '0', 
+        'timestamp': datetime.datetime(2021, 8, 25, 2, 51, 40, 891000, tzinfo=datetime.timezone.utc), 
+        'take_profit': '0', 
+        'stop_loss': '0', 
+        'trailing_stop': '0', 
+        'last_exec_price': '0', 
+        'reduce_only': False, 
+        'close_on_trigger': False}
+        """
```
Comment (boseok-bc) [inline cryptofeed/exchanges/bitmex.py]: Please replace this bybit example to bitmex example

File: cryptofeed/exchanges/bitmex.py [hunk 2]
```diff
@@ -59,6 +60,45 @@ def _parse_symbol_data(cls, data: dict) -> Tuple[Dict, Dict]:
...
+        if data.get('orderQty') is not None:
+            order['qty'] = data['orderQty']
+        if data.get('cumQty') is not None:
+            order['cum_exec_qty'] = data['cumQty']
+        if data.get('leavesQty') is not None:
+            order['leaves_qty'] = data['leavesQty']
+        if data.get('price') is not None:
+            order['price'] = data['price']
+        if (data.get('avgPrice') is not None) or (data.get('avgPx') is not None):
+            order['average'] = (data.get('avgPrice') or data.get('avgPx') or 0)
```
Comment (boseok-bc) [inline cryptofeed/exchanges/bitmex.py]: We have to generalize parsed order format.
for example, All exchanges have the same format of parsed orderbook.
bitmex.
https://github.com/block-crafters/cryptofeed/blob/94196ecfa85b939360adf1cafdd1737b888b8a48/cryptofeed/exchanges/bitmex.py#L108
okex
https://github.com/block-crafters/cryptofeed/blob/94196ecfa85b939360adf1cafdd1737b888b8a48/cryptofeed/exchanges/okex.py#L246-L253
ftx.
https://github.com/block-crafters/cryptofeed/blob/94196ecfa85b939360adf1cafdd1737b888b8a48/cryptofeed/exchanges/ftx.py#L299-L306
Like orderbook, order must also be in the same format.
I suggest following ccxt generalization. (They already consider about generalization very much)
https://github.com/ccxt/ccxt/blob/master/python/ccxt/bitmex.py#L1196-L1235 
In cryptofeed, _order methods of other exchanges(already implemented) is not considering generalization
For example, 
ftx - https://github.com/block-crafters/cryptofeed/blob/94196ecfa85b939360adf1cafdd1737b888b8a48/cryptofeed/exchanges/ftx.py#L364-L401
bybit - https://github.com/block-crafters/cryptofeed/blob/94196ecfa85b939360adf1cafdd1737b888b8a48/cryptofeed/exchanges/bybit.py#L345-L349
gemini - https://github.com/block-crafters/cryptofeed/blob/94196ecfa85b939360adf1cafdd1737b888b8a48/cryptofeed/exchanges/gemini.py#L121-L130 
These should be changed later.

**Review by boseok-bc — COMMENTED** [review]
Please check the comments! :)
And change the merge target branch to bc-develop from bc-master.

File: cryptofeed/exchanges/bitmex.py (hunk 2 above)
Comment (KunduzB) [reply cryptofeed/exchanges/bitmex.py]: @boseok-bc 

I have reviewed ccxt generalization of order.
So, I tried to include 'clientOrderId', 'stopPrice', 'postOnly', 'orderType' to the parsed order dict. However, when access them I get `TypeError: 'builtin_function_or_method' object is not subscriptable` .

Comment (boseok-bc) [reply cryptofeed/exchanges/bitmex.py]: The message is raised when you try to access builtin keyword.
ex) list['postOnly'] range['stopPrice']
Maybe there is this kind of a mistake in your code.

File: cryptofeed/exchanges/bitmex.py [hunk 3]
```diff
@@ -59,6 +60,73 @@ def _parse_symbol_data(cls, data: dict) -> Tuple[Dict, Dict]:
...
+        if data.get('ordType') is not None:
+            order['orderType'] = data.get('ordType')
+        if data.get('cost') is not None:
+            order['cost'] = data.get('cost')
+        if data.get('fee') is not None:
+            order['fee'] = data.get('fee')
+        if data.get('trades') is not None:
+            order['trades'] = data.get('trades')
+        if data.get('timeInForce') is not None:
+            order['timeInForce'] =  Bitmex.parse_time_in_force(data.get('timeInForce'))
```
Comment (boseok-bc) [inline cryptofeed/exchanges/bitmex.py]: 2 spaces :)

**Review by boseok-bc — APPROVED** [review]
Please remove a space and merge this PR.
Great 👍

File: cryptofeed/exchanges/bitmex.py [hunk 4]
```diff
@@ -59,6 +60,73 @@ def _parse_symbol_data(cls, data: dict) -> Tuple[Dict, Dict]:
...
+            order['side'] = BUY if data['side'] == 'Buy' else SELL
+        if data.get('ordStatus'):
+            order['status'] = Bitmex.parse_order_status(data['ordStatus'])
+
+        # 0 should be True at `if` statement.
+        if data.get('orderQty') is not None:
+            order['amount'] = Decimal(data['orderQty'])
+        if data.get('cumQty') is not None:
+            order['filled'] = Decimal(data['cumQty'])
+        if data.get('leavesQty') is not None:
```
Comment (hotag-vero) [inline cryptofeed/exchanges/bitmex.py:79]: 'remaining' is good. (rather than 'leaves')

File: cryptofeed/exchanges/bitmex.py [hunk 5]
```diff
@@ -59,6 +60,73 @@ def _parse_symbol_data(cls, data: dict) -> Tuple[Dict, Dict]:
...
+        }
+
+        ret = statuses.get(status, None)
+        return ret
+
+    @staticmethod
+    def parse_time_in_force(timeInForce):
+        times = {
+            'Day': 'Day',
+            'GoodTillCancel': 'GTC',
```
Comment (hotag-vero) [inline cryptofeed/exchanges/bitmex.py:123]: It's actually 'Good-Til-Cancelled' but looks okay.

File: cryptofeed/exchanges/bitmex.py [hunk 6]
```diff
@@ -59,6 +60,73 @@ def _parse_symbol_data(cls, data: dict) -> Tuple[Dict, Dict]:
...
+        if data.get('orderQty') is not None:
+            order['amount'] = Decimal(data['orderQty'])
+        if data.get('cumQty') is not None:
+            order['filled'] = Decimal(data['cumQty'])
+        if data.get('leavesQty') is not None:
+            order['remaining'] = Decimal(data['leavesQty'])
+        if data.get('price') is not None:
+            order['price'] = Decimal(data['price'])
+        if data.get('avgPx') is not None:
+            order['average'] = Decimal(data.get('avgPx'))
```
Comment (hotag-vero) [inline cryptofeed/exchanges/bitmex.py:84]: Perhaps 'avgPrice'?

**Review by hotag-vero — COMMENTED** [review]
Simple comments. Not so critical.

**Pull request merged**